core/core-db-repositories/src/main/java/com/sflpro/notifier/db/repositories/repositories/notification/push/impl/PushNotificationRecipientRepositoryImpl.java

```diff
@@ -0,0 +1,120 @@
+        // Execute query
+        final Long customersCount = typedQuery.getSingleResult();
+        LOGGER.debug("Successfully executed push notification recipients count lookup request for search parameters, parameters - {}", parameters);
+        return customersCount;
+    }
+
+    @Nonnull
+    @Override
+    public List<PushNotificationRecipient> findPushNotificationRecipients(@Nonnull final PushNotificationRecipientSearchFilter parameters, final long startFrom, final int maxCount) {
+        assertSearchParametersNotNull(parameters);
+        Assert.isTrue(maxCount > 0, "Max count should be positive integer");
+        Assert.isTrue(startFrom >= 0, "Start from should be positive integer or zero");
+        LOGGER.debug("Executing push notification recipients search request for search parameters, parameters - {}, startFrom - {} ,maxCount - {}", parameters, startFrom, maxCount);
+        // Build query
+        final TypedQuery<PushNotificationRecipient> typedQuery = buildFindRecipientsForSearchParametersTypedQuery(parameters, false, PushNotificationRecipient.class);
+        // Set pagination data
+        typedQuery.setFirstResult((int) startFrom);
+        typedQuery.setMaxResults(maxCount);
+        // Execute query
+        final List<PushNotificationRecipient> recipients = typedQuery.getResultList();
+        LOGGER.debug("Successfully executed recipients lookup request for search parameters, parameters - {}, startFrom - {} ,maxCount - {}", parameters, startFrom, maxCount);
+        return recipients;
+    }
+
+    /* Utility methods */
+    private void assertSearchParametersNotNull(final PushNotificationRecipientSearchFilter parameters) {
+        Assert.notNull(parameters, "Search parameters should not be null");
+    }
+
+    private <T> TypedQuery<T> buildFindRecipientsForSearchParametersTypedQuery(final PushNotificationRecipientSearchFilter parameters, final boolean countQuery, final Class<T> queryResultType) {
+        final String queryString = buildFindRecipientsForSearchParametersQueryString(parameters, countQuery);
+        // Create typed query
+        final TypedQuery<T> typedQuery = entityManager.createQuery(queryString, queryResultType);
+        typedQuery.setParameter(PARAMETER_NAME_SUBSCRIPTION_ID, parameters.getSubscriptionId());
+        typedQuery.setParameter(PARAMETER_NAME_PROVIDER_TYPE, parameters.getProviderType());
+        typedQuery.setParameter(PARAMETER_NAME_DEVICE_OPERATING_SYSTEM, parameters.getDeviceOperatingSystemType());
+        typedQuery.setParameter(PARAMETER_NAME_DESTINATION_ROUTE_TOKEN, parameters.getDestinationRouteToken());
+        typedQuery.setParameter(PARAMETER_NAME_STATUS, parameters.getStatus());
+        typedQuery.setParameter(PARAMETER_NAME_APPLICATION_TYPE, parameters.getApplicationType());
+        return typedQuery;
+    }
+
+    private String buildFindRecipientsForSearchParametersQueryString(final PushNotificationRecipientSearchFilter parameters, final boolean countQuery) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "parameters". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)